backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/picking/pickingslot/process/WEBUI_Picking_PickQtyToExistingHU.java

```diff
@@ -134,20 +130,24 @@ public Object getParameterDefaultValue(@NonNull final IProcessDefaultParameter p
 	protected void validatePickingToHU()
 	{
 		final PickingSlotRow pickingSlotRow = getSingleSelectedRow();
-
 		final I_M_HU hu = handlingUnitsBL.getById(pickingSlotRow.getHuId());
 
-		if (handlingUnitsBL.isVirtual(hu))
+		if (!handlingUnitsBL.isVirtual(hu))
 		{
-			throw new AdempiereException(msgBL.getTranslatableMsgText(MSG_WEBUI_PICKING_TO_EXISTING_CUS_NOT_ALLOWED));
+			return;
 		}
 
-		final OrderId orderIdOfThePickedHU = extractCorrespondingOrderId(pickingSlotRow.getHuId());
-		final OrderId orderIdOfTheCurrentSchedule = OrderId.ofRepoIdOrNull(getCurrentShipmentSchedule().getC_Order_ID());
+		final ImmutableSet<ProductId> productIds = handlingUnitsBL.getStorageFactory().streamHUProductStorages(ImmutableList.of(hu))
+				.map(IProductStorage::getProductId)
+				.collect(ImmutableSet.toImmutableSet());
+
+		final I_M_ShipmentSchedule selectedShipmentSchedule = getCurrentShipmentSchedule();
 
-		if (!orderIdOfThePickedHU.equals(orderIdOfTheCurrentSchedule))
+		if (productIds.size() != 1 || !productIds.contains(ProductId.ofRepoId(selectedShipmentSchedule.getM_Product_ID())))
 		{
-			throw new AdempiereException(msgBL.getTranslatableMsgText(MSG_WEBUI_PICKING_TO_THE_SAME_HU_FOR_MULTIPLE_ORDERS_NOT_ALLOWED));
+			throw new AdempiereException("VHUs cannot have multiple product storages!")
```

Comment:
i think this error message applies only for productIds.size() != 1.
For the other part of the OR we need a different error message.